src/components/implementation/fork_tests/voter/voter.c

```diff
@@ -89,17 +92,20 @@ int nwrite(spdid_t spdid, replica_type to, size_t sz) {
 		return -1; // whatever, shouldn't happen if blocked
 	}
 	else {
+		assert(replica->buf_write);
+		assert(sz <= 1024);
+		
 		replica->destination = to_comp;
+		replica->sz_write = sz;
 		printc("resplica destination set to %x\n", replica->destination);
-		assert(replica->buf_write);
 		ret = block_replica(replica);
 		if (ret < 0) printc("Error\n");
 		/* On wakeup */	
 		return 0;
 	}
 }
 
-int nread(spdid_t spdid, replica_type from, size_t sz) {
+size_t nread(spdid_t spdid, replica_type from, size_t sz) {
```

Comment:
A comment would be helpful and it's super hypocritical of me to not include one - you're right! Although I will put it in voter.h.

size_t sz is intended to tell the voter the maximum bytes the calling component _thinks_ it can read. Then the size_t return value is how many bytes were actually returned. I think this is inspired by POSIX? I've seen it somewhere before, and I believe Gabe suggested this unless I misunderstood.